azure/services/routetables/client.go

```diff
@@ -54,43 +50,112 @@ func newRouteTablesClient(subscriptionID string, baseURI string, authorizer auto
 }
 
 // Get gets the specified route table.
-func (ac *azureClient) Get(ctx context.Context, resourceGroupName, rtName string) (network.RouteTable, error) {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "routetables.AzureClient.Get")
+func (ac *azureClient) Get(ctx context.Context, spec azure.ResourceSpecGetter) (interface{}, error) {
+	ctx, _, done := tele.StartSpanWithLogger(ctx, "routetables.azureClient.Get")
 	defer done()
 
-	return ac.routetables.Get(ctx, resourceGroupName, rtName, "")
+	return ac.routetables.Get(ctx, spec.ResourceGroupName(), spec.ResourceName(), "")
 }
 
-// CreateOrUpdate create or updates a route table in a specified resource group.
-func (ac *azureClient) CreateOrUpdate(ctx context.Context, resourceGroupName string, rtName string, rt network.RouteTable) error {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "routetables.AzureClient.CreateOrUpdate")
+// CreateOrUpdateAsync creates or updates a route table asynchronously.
+// It sends a PUT request to Azure and if accepted without error, the func will return a Future which can be used to track the ongoing
+// progress of the operation.
+func (ac *azureClient) CreateOrUpdateAsync(ctx context.Context, spec azure.ResourceSpecGetter, parameters interface{}) (interface{}, azureautorest.FutureAPI, error) {
```

Comment:
Great suggestion. I just pushed an update for this & applied the change to other services that are already refactored as well. I kept the commit separate since it's a bit of a scope creep and required renaming some variables here and there.